sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdClientChannelPool.java

```diff
@@ -58,6 +58,44 @@
 
 /**
  * A {@link ChannelPool} implementation that enforces a maximum number of concurrent direct TCP Cosmos connections.
+ *
+ * RntbdClientChannelPool: Actors
+ * 	- acquire (RntbdServiceEndpoint): acquire a channel to use
+ * 	- release (RntbdServiceEndpoint): channel usage is complete and returning it back to pool
+ * 	- Channel.closeChannel() Future: Event handling notifying the channel termination to refresh bookkeeping
+ * 	- acquisitionTimeoutTimer: channel acquisition wait-out handler
+ * 	- monitoring (through RntbdServiceEndpoint): get monitoring metrics
+ *
+ * 	Behaviors/Expectations:
+ * 	    - Bounds: MAX_CHANNELS_PER_ENDPOINT * MAX_REQUESTS_ENDPOINT (NOT A GUARANTEE)
```

Comment:
This is my understanding, lets please validate.

Reply:
This coupled with serial connection creation might be a possible reason for high start-up latency at load for a single endpoint if-any.  Thoughts?

Reply:
- Bounds:
    - PendingAcquisition queue: default --> Max(10_000, MAX_CHANNELS_PER_ENDPOINT * MAX_REQUESTS_ENDPOINT)
    - Executor's event loop task queue: MAX_CHANNELS_PER_ENDPOINT * MAX_REQUESTS_ENDPOINT
    - AvailableChannels.size()  + AcquiredChannels.size() <= MAX_CHANNELS_PER_ENDPOINT

Reply:
Each acquisition also will be queued right, max tasks = #ENDPOINTS * MAX_CHANNELS_PER_ENDPOINT * MAX_REQUESTS_ENDPOINT + newAcquisitions (not yet in pendingAcquisitions)